workflows_api/runtime/src/auth.py

```diff
@@ -18,56 +26,37 @@ def get_settings() -> config.Settings:
     return main.settings
 
 
-def get_jwks_url(settings: config.Settings = Depends(get_settings)) -> str:
-    import boto3
-    import json
-    client = boto3.client("secretsmanager")
-    response = client.get_secret_value(SecretId=settings.workflows_client_secret_id)
-    secrets = json.loads(response["SecretString"])
-
-    return f"https://cognito-idp.{secrets['aws_region']}.amazonaws.com/{secrets['userpool_id']}/.well-known/jwks.json"
-
-
-@cached(TTLCache(maxsize=1, ttl=3600))
-def get_jwks(jwks_url: str = Depends(get_jwks_url)) -> KeySet:
-    with requests.get(jwks_url) as response:
-        response.raise_for_status()
-        return JsonWebKey.import_key_set(response.json())
-
-
-def decode_token(
-    token: security.HTTPAuthorizationCredentials = Depends(token_scheme),
-    jwks: KeySet = Depends(get_jwks),
-) -> JWTClaims:
-    """
-    Validate & decode JWT
-    """
+def validated_token(
+    token_str: Annotated[str, Security(oauth2_scheme)],
+    required_scopes: security.SecurityScopes,
+):
```

Comment:
Can you include type hint?

Reply:
https://github.com/NASA-IMPACT/veda-data-airflow/pull/138/commits/c4f9829bb06b7c047dee8f62cc58a58dd68c0f34